Laboratory work 4

Ateus1 opened the pull request.

Ateus1 committed:
Laboratory work 4

rmushkin commented on Laboratory work 4/task_1.py:
```diff
@@ -0,0 +1,73 @@
+        Данный метод необходим, чтобы предотвратить возникновение ошибки,
+        если в реализации дочернего класса будет отсутствовать метод вычисления площади.
+        """
+        return 0
```
Выбрасывайте исключение `NotImplementedError`, так будет правильнее (вы как бы заставите имплементировать этот метод в дочерних классах)

rmushkin commented on Laboratory work 4/task_1.py:
```diff
@@ -0,0 +1,73 @@
+
+    def __repr__(self) -> str:
+        """Возвращает информацию о фигуре в режиме отладки."""
+        return f"{self.__class__}: {self.name} (цвет: {self.color}, площадь: {self.area()}, вершин: {self.vertices})"
```
`__repr__` обычно делают "фотографией" вызова метода-конструктора. Т. е. оно долно вернуть что-то типа `Shape(color=..., name=..., vertices=...)`

rmushkin reviewed (changes requested):
Требуется исправление

Ateus1 committed:
Исправлены, указанные замечания.

rmushkin reviewed (changes requested):
Ещё немного.

Ateus1 committed:
Исправлены ошибки.

1. Метод area в базовом классе Shape, теперь не возвращает значение NotImplementedError, а выбрасывает ошибку NotImplementedError.
2. В дочернем классе Rectangle метод __repr__, теперь перегружен, а не наследуется от родительского класса Shape.

rmushkin reviewed (approved):
Принимается ✅
